### PR TITLE
check path equivalency

### PR DESCRIPTION
**Pull request by HiteshRepo**
This change is to address the path related issue that comes up while `corso repo connect filesystem`.
Below are the steps to reciprocate:
1. Initialize repo with filesystem provider (Notice that the path has a trailing slash): 
command: `./corso repo init filesystem --path ~/Documents/Backups/`
output:
Logging to file: ~/.cache/corso/logs/2023-11-09T05-21-57Z.log
Connecting to M365:         done 
Initializing repository   
Initialized a repository at path ~/Documents/Backups/

2. Connect repo with filesystem provider:
command: `./corso repo connect filesystem --path ~/Documents/Backups/`
output:
Logging to file: ~/.cache/corso/logs/2023-11-09T05-23-30Z.log
Error: retrieving storage provider details: verifying storage configs in corso config file: verifying configs in corso config file: value of path (~/Documents/Backups) does not match corso configuration value (~/Documents/Backups/)

#### Does this PR need a docs update or release note?

- [ ] :white_check_mark: Yes, it's included
- [ ] :clock1: Yes, but in a later PR
- [x] :no_entry: No

#### Type of change

- [ ] :sunflower: Feature
- [x] :bug: Bugfix
- [ ] :world_map: Documentation
- [ ] :robot: Supportability/Tests
- [ ] :computer: CI/Deployment
- [ ] :broom: Tech Debt/Cleanup

#### Test Plan

- [x] :muscle: Manual
- [x] :zap: Unit test
- [x] :green_heart: E2E

### CODE REVIEW

File: src/pkg/path/path.go
Comment (ryanfkeepers) [inline src/pkg/path/path.go:322]: could we get some unit tests for this func before merging, please?

File: src/pkg/path/path.go
Comment (ryanfkeepers) [inline src/pkg/path/path.go:332]: The name is a bit misleading, since many inputs to this func could hold valid paths without passing the stat check.  This name makes it a bit more clear what we're checking.
```suggestion
func DoesPathExistInOS(path string) bool {
```

File: src/pkg/storage/storage.go
Comment (ryanfkeepers) [inline src/pkg/storage/storage.go:179]: It should be possible to skip the valid check and call `path.ArePathsEquivalent` for any pair of values, right?  Whether or not they clear os.Stat doesn't actually change how ArePathsEquivalent compares them.

**Review by ryanfkeepers — APPROVED** [review]
Some nits on design choices to evaluate, but nothing blocking.  Approving to unblock while other details get finalized.

File: src/pkg/storage/storage.go
Comment (HiteshRepo) [reply src/pkg/storage/storage.go:179]: The value pairs can be any string other than paths. So the idea was to compare paths only if they qualify as paths.

File: src/pkg/storage/storage.go
Comment (pandeyabs) [inline src/pkg/storage/storage.go:175]: Could you please add a comment to describe this code addition? Also once concern I have is that there are non path related keys in the map. This code seems to be doing `os.stat()` call for  all the keys. One could argue that these could be valid file paths on a user's system, could be interpreted incorrectly as a snooping behavior.

Is there a way we could only do this check for file system path key? Btw, thanks for fixing the trailing slash bug! Good find!

Comment (HiteshRepo) [reply src/pkg/storage/storage.go]: @pandeyabs please check the current change. Do you concur with the current implementation?

**Pull request merged**
